proxy/envoy/testdata/envoy.json.golden

```diff
@@ -0,0 +1,286 @@
+{
```

Comment:
How was this file generated and how would it be refreshed/updated, e.g. new envoy features, schema changes? Maybe add a smaller helper program which uses the mocks you added but just saves the JSON to a file and skips any validation step?